bin/python_modules/integronfinder_process.py

```diff
@@ -9,42 +9,64 @@
-                    if gbk_file in inf_gbks:
-                        flag = 0
-                        for gb_record in SeqIO.parse(gbk_file, "genbank"):
-                            for feature in gb_record.features:
-                                if feature.type == "integron":
-                                    if (
-                                        feature.qualifiers["integron_type"][0]
-                                        == "complete"
-                                    ):
-                                        flag = 1
-                                        mge_counter += 1
-                                        mge_id = "inf_" + str(mge_counter)
-                                        start = int(feature.location.start)
-                                        end = int(feature.location.end)
-                                        coord = (start, end)
-                                        value = (id_replicon, description, coord)
-                                        mge_data[mge_id] = value
-                                if feature.type == "attC":
-                                    if flag == 1:
-                                        start = str(feature.location.start)
-                                        end = str(feature.location.end)
-                                        coord = (start, end)
-                                        attC_site[mge_id] = coord
-                                        flag = 0
+                    gbk_files_list.append(gbk_file)
+
+
+    mge_counter = 0
+    attC_site = {}
+    for gbk_file in gbk_files_list:
+        for gb_record in SeqIO.parse(gbk_file, "genbank"):
+
+            # Get indexes for integron records and attC
+            indexes_integron, indexes_attc = [], []
+            for index, feature in enumerate(gb_record.features):
+                if feature.type == "integron":
+                    indexes_integron.append(index)
+                elif feature.type == "attC":
+                    indexes_attc.append(index)
+
+            # process all integrons
+            for index_integron in indexes_integron:
+                integron_feature = gb_record.features[index_integron]
+                if integron_feature.qualifiers["integron_type"][0] == "complete":
+                    mge_counter += 1
+                    mge_id = "inf_" + str(mge_counter)
+                    mge_start = int(integron_feature.location.start)
+                    if mge_start == 0:
+                        mge_start = 1
+                    mge_end = int(integron_feature.location.end)
+                    mge_coord = (mge_start, mge_end)
+                    description = "mobile_element_type=complete_integron"
+                    id_replicon = gbk_file.replace('.gbk','')
+                    value = (id_replicon, description, mge_coord)
+                    mge_data[mge_id] = value
+
+                    print('process integron -> check attC')
```

Comment:
do you need prints here? maybe that was just for testing :) but leave if they are necessary

Reply:
removed!